[v1.3][W14-B2] PTMan

shanwpf opened the pull request:
@koo1993 @sunbangjie @hzxcaryn

okkhoy commented on README.adoc:
```diff
@@ -1,11 +1,9 @@
-https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master[Coverage Status]]
-https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&utm_medium=referral&utm_content=se-edu/addressbook-level4&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a[Codacy Badge]]
-https://gitter.im/se-edu/Lobby[image:https://badges.gitter.im/se-edu/Lobby.svg[Gitter chat]]
+https://travis-ci.org/CS2103JAN2018-W14-B2/main[image:https://travis-ci.org/CS2103JAN2018-W14-B2/main.svg?branch=master[Build Status]]
```
Please update the title of your project! It is not AddressBook anymore.

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -798,35 +798,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  Employer enters employee details and working outlet(s)
```
Maybe you can reword this `Employer enters employee details and working outlets to add employee to the system`

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -840,18 +904,124 @@ Use case ends.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should respond within 2 seconds after entering any command.
+.  Should be usable by users who has never used a command line interface before.
+.  Should be expected to manage contact details of part-time workers in any outlet.
```
should be `able` to...

okkhoy commented on src/main/java/seedu/address/model/AddressBook.java:
```diff
@@ -155,6 +156,58 @@ public void addTag(Tag t) throws UniqueTagList.DuplicateTagException {
+     * @param tag
+     * @param person
+     */
+    private void removeTagFromPerson(Tag tag, Person person) {
```
There is no update to the user guide for this feature. Remember, any new feature or change in functionality should ALWAYS be accompanied by an update to the tests and documentation.

okkhoy commented on src/main/java/seedu/address/storage/XmlAddressBookStorage.java:
```diff
@@ -79,4 +80,22 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) th
     }
 
+    @Override
+    public void backupAddressBook(ReadOnlyAddressBook addressBook) throws IOException {
```
Same comment as above!

okkhoy reviewed (commented):
I have a feeling that the scope of the user stories recorded is quite big. You may want to re-prioritize on the deliverables.
Good job on this!

okkhoy commented:
@shanwpf @koo1993 @SunBangjie @hzxcaryn

SunBangjie committed:
update UG, add commands related to Outlet and Shift

waynekoo committed:
Update userGuide once again (#53)

* update AboutUs.adoc

* update User Guide for Salary

* getting ready for V2.0 userGuide

* minor change in wording and sequence in userguide

* update userGuide and merge conflict

SunBangjie committed:
update outlet information

hzxcaryn committed:
Update User Guide (Formatting and English)

hzxcaryn committed:
Tiny modifications to user guide

waynekoo committed:
Update developer guide (#14)

waynekoo committed:
Access control (#58)

-now PTManger holds a password and all employee are given default password appoint creation.
-commands now follow closely to USER GUIDE.

ADMINPASSWORD FOR THIS BUILD IS: `DEFAULT1`

shanwpf committed:
Add feature contribution to developer guide

shanwpf committed:
Update DeveloperGuide.adoc

SunBangjie committed:
add command editoutlet

SunBangjie committed:
add test

waynekoo committed:
Access control new structure! (#61)

* Add password class and test case for password

* update password class and test to fix error in automated tests

* update test for coverage

* update test for coverage

* implement password for add command.

* update new structure for access control using password

* Enhance code and fix multiple bug

* update Test cases to fit new command structure

* fix coverage and add test cases

* UPDATE accress control structure.

* update minor bug fix

* test case to fit the new access control structure

* Merge branch 'master' of https://github.com/CS2103JAN2018-W14-B2/main into AccessControl

# Conflicts:
#	src/main/java/seedu/ptman/logic/commands/AddCommand.java
#	src/main/java/seedu/ptman/logic/commands/DeleteCommand.java
#	src/main/java/seedu/ptman/logic/commands/EditCommand.java
#	src/main/java/seedu/ptman/logic/commands/RedoCommand.java
#	src/main/java/seedu/ptman/logic/commands/UndoCommand.java
#	src/main/java/seedu/ptman/logic/parser/AddCommandParser.java
#	src/main/java/seedu/ptman/logic/parser/DeleteCommandParser.java
#	src/main/java/seedu/ptman/logic/parser/PartTimeManagerParser.java
#	src/main/java/seedu/ptman/model/Model.java
#	src/main/java/seedu/ptman/model/ModelManager.java
#	src/main/java/seedu/ptman/model/PartTimeManager.java
#	src/main/java/seedu/ptman/model/Password.java
#	src/test/java/seedu/ptman/logic/commands/AddCommandTest.java
#	src/test/java/seedu/ptman/logic/parser/AddCommandParserTest.java

* update clear command to execute when provided password

* update test case for clear,redo,undo command parser

* update test

waynekoo committed:
update contribution in developer guide

okkhoy commented on README.adoc:
```diff
@@ -15,25 +13,27 @@ ifndef::env-github[]
+* For employees:
+. View the available working time slots through the timetable.
+. Register for preferred time slots that you want to work at.
+* A sophisticated GUI that includes a list panel and an in-built timetable.
```
Just a question, do you manage allotment of employees to time slots as well?

okkhoy commented on README.adoc:
```diff
@@ -15,25 +13,27 @@ ifndef::env-github[]
 == Acknowledgements
 
-* Some parts of this sample application were inspired by the excellent http://code.makery.ch/library/javafx-8-tutorial/[Java FX tutorial] by
+* Some parts of this application were inspired by the excellent http://code.makery.ch/library/javafx-8-tutorial/[Java FX tutorial] by
```
Need to attribute SE-EDU addressbook-level4 as well

waynekoo replied:
thanks for the input we updated the readme

okkhoy commented on docs/AboutUs.adoc:
```diff
@@ -3,53 +3,53 @@
 
 Role: Developer +
-Responsibilities: UI
+Responsibilities: Integration + Model component + Documentation
```
Can you indicate one of you as the team lead?

waynekoo replied:
we have indicated the team lead :D

okkhoy commented on docs/UserGuide.adoc:
```diff
@@ -11,226 +12,341 @@ ifdef::env-github[]
+==== Adding an employee: `add`
+
+Adds an employee to PTMan. +
+*Format:* `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS s/SALARY [t/TAG]... pw/ADMIN_PASSWORD` +
```
1. Do you mean that for every employee to be added, the employer has to enter his password? If so, think how you can make employers' life simple

2. Are you asking for the password in plain text? :confused:

okkhoy commented on docs/UserGuide.adoc:
```diff
@@ -11,226 +12,341 @@ ifdef::env-github[]
-Edits an existing person in the address book. +
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]...`
+[NOTE]
+An employer password can't be edited. [refer to `rp` (reset password)]
```
I can see `rp` only resets employee password. What about the employer? Can't be edited is acceptable, but can't be reset is a major gap!

waynekoo replied:
we rethought through and came out with a solution that allows both employee and employer to reset their password. I will be reflected in user guide.

okkhoy commented on docs/UserGuide.adoc:
```diff
@@ -11,226 +12,341 @@ ifdef::env-github[]
-`undo` (reverses the `clear` command) +
-`undo` (reverses the `delete 1` command) +
+* `delete 1 pw/ADMIN_PASSWORD` +
+`clear pw/ADMIN_PASSWORD` +
```
I think asking for the password to be entered everytime is a hassle + security issue. You may have someone peeping into your screen! :stuck_out_tongue_winking_eye:

okkhoy reviewed (commented):
Good job guys! I see this as a comprehensive update!

May I suggest adding something to be done in the future (v2.0)?

okkhoy commented:
@shanwpf @SunBangjie @hzxcaryn @koo1993 

Please keep up the good work!

I would want you to mention the following in your **Developer guide**
1. For each team member what are the features (major and minor) you are proposing?
2. Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

SunBangjie committed:
update testutil

SunBangjie committed:
add test

SunBangjie committed:
add viewoutlet command for demo purpose

SunBangjie committed:
Merge branch 'master' into EditOutlet

SunBangjie committed:
fix coding style

SunBangjie committed:
Merge branch 'EditOutlet' of https://github.com/SunBangjie/main into EditOutlet

waynekoo committed:
Revert "Access control new structure! (#61)"

This reverts commit dc18c599e038886019d05f47dfb7d5712c843711.

shanwpf committed:
Create and implement AddShiftCommand (#57)

shanwpf committed:
Create and implement delete shift command (#64)

Format: `deleteshift <index>`
Alias: `ds`

SunBangjie committed:
update contribution

SunBangjie committed:
Merge branch 'master' into EditOutlet

SunBangjie committed:
Merge pull request #66 from SunBangjie/UpdateDG

Update DG

SunBangjie committed:
fix and add tests

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
Merge pull request #60 from SunBangjie/EditOutlet

Create and implement editoutlet command

hzxcaryn committed:
Update DG with feature contri (#67)

waynekoo committed:
update new structure for access control (#65)

* update new structure for access control

* update test for new access control

* update userGuide for the new access control

* Integrate with shannon's code

* update test for access control

* update tests

* Integration with bangjie's code

* integrate with outletinformation to access password.

waynekoo committed:
update developer guide doc

SunBangjie committed:
add Contact to OutletInformation

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
add tests

SunBangjie committed:
fix coding style

waynekoo committed:
update about us doc

SunBangjie committed:
fix coding style

waynekoo committed:
Update README.adoc

SunBangjie committed:
add test

waynekoo committed:
Update DeveloperGuide.adoc

SunBangjie committed:
delete trailing whitespace

SunBangjie committed:
Merge pull request #72 from SunBangjie/AboutUs

Delete trailing whitespace in AboutUs

SunBangjie committed:
edit tests

SunBangjie committed:
Merge pull request #71 from SunBangjie/UpdateOutletInformation

Add Contact to OutletInformation

SunBangjie committed:
update undo editoutlet command

shanwpf committed:
Update developer guide to include Shift implementation (#74)

SunBangjie committed:
Add test

SunBangjie committed:
Merge pull request #73 from SunBangjie/UndoEditOutletCommand

Update undo editoutlet command

SunBangjie committed:
update in DG implementation

hzxcaryn committed:
UI modifications (Outlet Panel and Timetable GUI) (#54)

* Modify UI of employee list panel with header

* Add Employee Panel header

* Modify layout of main window

* Change app title from Address App to PTMan

* Edit result display style

* Add Outlet details panel and timetable view to Main Window

* Modify result display default text

* Remove BrowserPanel related tests

* Update theme css

* Add ChangedEvent classes for outlet name and information

* Remove more browser panel tests

* Update handles (main window, outlet details panel, timetable view)

* add outlet details and timetable view handle

* Add Calendarfx libs

* Add libs dependency to gradle for calendarFX

* Add relevant changes for timetable

* Update TimetablePanel class (and a name change)

* Update relevant parts after name change (timetableView to timetablePanel)

* Implement timetable view using CalendarFX

* Start timetable view week from Monday

* Add simple test for timetable panel

* Remove unused method in Timetable

* Add Outlet Panel UI tests

* Add Outlet Panel UI test

SunBangjie committed:
Merge pull request #75 from SunBangjie/UpdateDGImplementation

Update in DG implementation

waynekoo committed:
update implementation (#79)

#49

hzxcaryn committed:
Documentations (Update readme and DG) (#80)

* Update readme

* Update DG with timetable feature

* correct typo

SunBangjie committed:
add email in outlet information

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
Merge pull request #83 from SunBangjie/AddEmail

Add email in outlet information

shanwpf committed:
Fix redo for outlet and shift commnads (#93)

SunBangjie committed:
display default outlet information

SunBangjie committed:
add tests and fix logic

SunBangjie committed:
clean up code

SunBangjie committed:
clean up code

SunBangjie committed:
display in UI

SunBangjie committed:
fix coding style

SunBangjie committed:
clean up code

SunBangjie committed:
Merge pull request #94 from SunBangjie/ReflectChangeInUI

Reflect outlet information change in UI

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -230,17 +240,116 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+The following commands directly interact with the `Shift` class:
+
+. `AddShiftCommand`: Creates a `Shift` and adds it to the `UniqueShiftList` in `OutletInformation`.
```
Consider using a class diagram to depict these details

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -230,17 +240,116 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+`TimetablePanel` takes in an `ObservableList<Shift>` object, which is an unmodifiable sorted view of the filtered shifts.
+ `TimetablePanel` then uses these `Shift` objects to set the Entries in the timetable view, in the method `setShifts()`.+
+
+The timetable GUI automatically refreshes to update the timetable view by calling the `updateTimetableView()` method whenever
```
How does auto-refresh work? You can show that in a SD.

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,30 +458,141 @@ image::UndoRedoActivityDiagram.png[width="650"]
+----
+This default constructor initializes the five fields of `OutletInformation` to respective default values:
+----
+    public OutletInformation() {
```
Please use code snippets minimally.
Reason: it is difficult to maintain these in the document, particularly if the implementation changes.

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,30 +458,141 @@ image::UndoRedoActivityDiagram.png[width="650"]
+----
+OutletInformation(OutletName outletName, OperatingHours operatingHours, OutletContact outletContact, Timetable timetable)
+----
+This constructor allows flexity in storing outlet information, but master password must be set to the default value.
```
flexity? did you mean flexibility?

okkhoy reviewed (commented):
Some of the features' implmentations are well described while others are too brief. Please update such feature implementation details.

Please use various UML diagrams to explain your implementation.

okkhoy commented:
@shanwpf @koo1993 @SunBangjie @hzxcaryn 
Some comments added to the DG. Please take a look and update as necessary.

SunBangjie committed:
store outlet info

SunBangjie committed:
modify tests

SunBangjie committed:
fix coding style

SunBangjie committed:
fix coding style

SunBangjie committed:
add announcement

SunBangjie committed:
fix coding style

SunBangjie committed:
modify viewoutlet command to show more message

SunBangjie committed:
fix coding style

SunBangjie committed:
add java doc

SunBangjie committed:
add test to increase coverage

SunBangjie committed:
fix test

SunBangjie committed:
fix coding style

SunBangjie committed:
add tests to increase coverage

SunBangjie committed:
fix import order

SunBangjie committed:
add tests to increase coverage

SunBangjie committed:
Merge pull request #96 from SunBangjie/StoreOutletInfo

Store outlet info

SunBangjie committed:
update UGDG

SunBangjie committed:
Merge pull request #99 from SunBangjie/UpdateUGDG

update UGDG

hzxcaryn committed:
UI revamp (#98)

* Remove irrelevant stuff from timetable view

* Add a tooltip to help the lost souls

* Update timetable panel initialisation

* Update DG to match timetable changes

* Add Shift index to timetable view

* Clean up code

* Update timetable panel test

* Minor changes to fxml

* Edit outlet details panel layout

* Edit min width of timetable

* Add more ui tests for outlet announcements

* Fix travis by removing min width of timetable

hzxcaryn committed:
Update UG for select and export commands

hzxcaryn committed:
Fix timetable view bug

shanwpf committed:
Create and implement apply command (#95)

Format: `apply EMPLOYEE_INDEX SHIFT_INDEX [/pw Password]
Alias: `ap`

waynekoo committed:
Changepw command for employee and employer (admin) (#101)

implement change password for employer and employee.
fixes #59

waynekoo committed:
update user story developer adoc

shanwpf committed:
Create and implement the unapply command (#108)

shanwpf committed:
Update documentation (#112)

hzxcaryn committed:
Add export timetable command (#111)

* Add export timetable command

* Add export command test

* Add parser tests

* Add test for equals

* Add export command parser test

* Add snapshot tests

* Improve code style

* Remove trailing whitespace

waynekoo committed:
Reset password for employer and employee (#113)

close #102

hzxcaryn committed:
Use open source libs

hzxcaryn committed:
Remove readme of libs (#118)

* Remove readme of libs

* Remove readme in libs

hzxcaryn committed:
Select command to show employee's shifts (#115)

* Add basic code for select command

* Timetable highlights according to shifts of selected employee

* Clean up

* Add timetable tests

* Update user stories

* Fix timetable test

waynekoo committed:
UPDATE UserGuide rp cmp rmp cp

hzxcaryn committed:
Add logo

hzxcaryn committed:
Update version

SunBangjie committed:
simple encryption and decryption

SunBangjie committed:
encrypt and decrypt employees

SunBangjie committed:
update text

SunBangjie committed:
fix coding style

hzxcaryn committed:
Add admin mode label

hzxcaryn committed:
Add tests for admin mode display

hzxcaryn committed:
Export Command to support emailing of exported image (#127)

* Allow export timetable to send email

* Modify tests for email

* Add comments

hzxcaryn committed:
Fix apply shift timetable bug

hzxcaryn committed:
Add main command (#130)

* Add main command

* Add timetable panel test for main command

* Update user guide

* Remove trailing whitespace

shanwpf committed:
Change shifts to use dates instead of days (#131)

Closes #131 
Closes #109

waynekoo committed:
Obscure password whenever user is typing on textfield (#132)

* obscure password when typing in textfield

resolve #126

waynekoo committed:
fixes #30 (#133)

shanwpf committed:
Save shifts to local storage (#134)

Closes #110

shanwpf committed:
Do not include password when checking employee equality (#135)

shanwpf committed:
Refactor classes and update comments (#136)

* `Timetable` class has been refactored into a `DateUtil` class.

* `Shift`, `Date`, `Time` and `Capacity` classes have been moved from `seedu.ptman.model.outlet` to `seedu.ptman.model.shift`. Likewise for their equivalent tests.

* Comments and annotations updated.

shanwpf committed:
Check if shift is full when adding employees to it (#139)

waynekoo committed:
Update history such that it will not display password in clear (#141)

* update history such that it will not store the password in clear text

shanwpf committed:
Update ApplyCommand to not require password in admin mode (#142)

shanwpf committed:
Update documentation (#140)

* Update user guide

* Update diagrams and developer guide

SunBangjie committed:
increase coverage

SunBangjie committed:
increase coverage

SunBangjie committed:
fix test bug

SunBangjie committed:
Merge pull request #121 from SunBangjie/Encrypter

Simple Encryption and Decryption

SunBangjie committed:
fix master password storage

SunBangjie committed:
Merge pull request #144 from SunBangjie/FixMasterPasswordStorage

Fix master password storage

SunBangjie committed:
encrypt shift

SunBangjie committed:
add test

SunBangjie committed:
fix coding style

SunBangjie committed:
Merge pull request #145 from SunBangjie/encryptShift

Encrypt shift in storage

SunBangjie committed:
update UGDG

SunBangjie committed:
Merge pull request #146 from SunBangjie/UpdateDocumentation

update UGDG

waynekoo committed:
update documentation and author annotation  (#147)

* update author of the code

* update UserGuide

SunBangjie committed:
Update author annotation

SunBangjie committed:
Merge pull request #148 from SunBangjie/AuthorAnnotation

Update author annotation

waynekoo committed:
refactor change admin password command and annotation for author

waynekoo committed:
refactor code

hzxcaryn committed:
New CSS

hzxcaryn committed:
Code cleanup

hzxcaryn committed:
timetable view to show capacity as well

hzxcaryn committed:
Add ptman logo icon to side of command box

hzxcaryn committed:
Change highlight color

hzxcaryn committed:
Add newline at EOF

hzxcaryn committed:
Edit whitespace

hzxcaryn committed:
Fix list cell

waynekoo committed:
update developer guide and minor typo

hzxcaryn committed:
Add author annotation

SunBangjie committed:
update DG

SunBangjie committed:
Merge pull request #155 from SunBangjie/DG

update DG

shanwpf committed:
Update documentation (#154)

* Update UGDG

* Add screenshots of shift indexing

waynekoo committed:
fix user able to see password by highlighting and funny behavior with the caret.

hzxcaryn committed:
Update UGDG (#156)

* Update UG with some images and guided examples

* Edit formatting of guided examples

* Fix merge conflicts

* Edit ug formatting

* Fix typos

* Replace mockup with real screenshot

* Clean up

* Update DG

* Correct english

waynekoo committed:
upload collate (#158)

hzxcaryn committed:
Use darker colours

shanwpf committed:
Remove an employee from all shifts when he is deleted from PTMan (#162)

Fixes #160

shanwpf committed:
Use figure references in docs (#159)

* Use figure references in DG

* Use figure references in UG

hzxcaryn committed:
Fix selection issue

hzxcaryn committed:
Add tests for deselecting

hzxcaryn committed:
Improve employee card pane (#166)

* Change color scheme

* Slight color changes

* Add more font styles

* Upgrade employee card

* Slight UI changes

shanwpf committed:
Update project portfolio (#167)

hzxcaryn committed:
Clean up commands, prep for autocompletion (#168)

SunBangjie committed:
fix OperatingHours bug

SunBangjie committed:
add tests

SunBangjie committed:
Merge pull request #170 from SunBangjie/fixBug

fix OperatingHours bug

shanwpf committed:
Documentation update (#171)

* Update project portfolio

* Change the user guide format

hzxcaryn committed:
More relevant sample util (#169)

shanwpf committed:
Update documentation (#172)

* Update project portfolio

* Change the user guide format

* Add TODO comments to UG

* Update DG

SunBangjie committed:
add encrypt/decrypt commands

SunBangjie committed:
Add tests

SunBangjie committed:
fix coding style

SunBangjie committed:
add tests to increase coverage

SunBangjie committed:
add tests

SunBangjie committed:
Merge pull request #193 from SunBangjie/EncryptDecryptCommand

Add encrypt/decrypt commands

SunBangjie committed:
update time input to HHMM format

SunBangjie committed:
fix test

SunBangjie committed:
Merge pull request #195 from SunBangjie/HHMM

update time input to HHMM format

hzxcaryn committed:
Add navigation of timetable week (#192)

* Add navigation of timetable week using keyboard keys Ctrl+Shift+Left/Right

* Add navbar for timetable with nav buttons

* Fix style

* Add date util tests

* Rename timetableview to timetablepanel

* Update UG

hzxcaryn committed:
Revert "Add navigation of timetable week (#192)" (#199)

This reverts commit f5ed8f73304805038027c6415c04e80a552eb254.

waynekoo committed:
Add salary command  (#198)

* add command for employer to increase salary

* update test ffor addSalaryCommand

* minor fix to code

* update collate

hzxcaryn committed:
Timetable navigation (#200)

* Add navigation of timetable week using keyboard keys Ctrl+Shift+Left/Right

* Add navbar for timetable with nav buttons

* Fix style

* Add date util tests

* Rename timetableview to timetablepanel

* Update UG

* Minor formatting

shanwpf committed:
Polish the Shifts feature (#201)

* Prevent adding of shifts to dates that are already over

* Prevent possible ArrayOutOfBoundsException with ApplyCommandParser

* Do not consider employees when checking shift equality

waynekoo committed:
update ppp and developer guide (#205)

* update developer Guide

* update developerGuide and ppp

* update ppp

SunBangjie committed:
backup files

SunBangjie committed:
add tests

SunBangjie committed:
add tests

SunBangjie committed:
Merge pull request #208 from SunBangjie/BackupFiles

Fix backup files

hzxcaryn committed:
List employees in shift (#207)

* Add viewshift command

* Update ug

* Add parser tests

* increase font size

* More parser tests

SunBangjie committed:
update DG

SunBangjie committed:
update DG

shanwpf committed:
Display an error message when a shift's start time is after end time (#204)

SunBangjie committed:
Update UG

SunBangjie committed:
update PPP

SunBangjie committed:
Merge pull request #212 from SunBangjie/UGDG

Update UGDG and Project Portfolio

shanwpf committed:
Do not allow users to apply for shifts that are over (#214)

shanwpf committed:
Update documentation (#215)

* Update project portfolio tag

* Update UserGuide.adoc

hzxcaryn committed:
Remove browser panel (#217)

hzxcaryn committed:
Update docs (#216)

waynekoo committed:
Update documentation and code base consistency 

* update user guide , update codebase to be more consistent

* update UserGuide image link

* update user guide

* update Test

* update xref to all ppp .adoc

shanwpf committed:
Fix detection of shifts that are over (#220)

* Fix detection of shifts that are over

* Set default date of shiftbuilder to be the next day from current date]

SunBangjie committed:
fix column issue

SunBangjie committed:
update about us

SunBangjie committed:
update DG

SunBangjie committed:
fix tests

SunBangjie committed:
fix typo

SunBangjie committed:
add test

SunBangjie committed:
add tests

SunBangjie committed:
Merge pull request #221 from SunBangjie/fix

Update DG and fix operating hours display issue

waynekoo committed:
v1.5 Clean Up and documentation (#222)

* update code quality

* update collate code

SunBangjie committed:
clean up code and tests

SunBangjie committed:
Merge pull request #223 from SunBangjie/master

Clean up code and tests on outlet feature

SunBangjie committed:
update UGDG

SunBangjie committed:
Merge pull request #224 from SunBangjie/Doc

update UGDG on outlet feature

waynekoo committed:
update PPP and with new image

hzxcaryn committed:
Rename main to defaultview (#218)

* Rename main to defaultview

* Rename main in UG

* Rename main in timetablepanel

* update ugdg

* Edit phrasing

* Split defaultviewcommand to deselect command and keyboard shortcut to move to current page

hzxcaryn committed:
fix left right bug for timetable (#226)

* Fix left and right issue

* Edit ug for ctrl/cmd+shift+down

hzxcaryn committed:
Add more employees and shifts into sample util (#227)

* Add sample util

* Fix indentation

* Update test storage

* Minor tweak

waynekoo committed:
update ppp

waynekoo committed:
update collate

SunBangjie committed:
update UGDG and PPP

SunBangjie committed:
add newline at EOF

SunBangjie committed:
Merge pull request #230 from SunBangjie/document

update UGDG and PPP

SunBangjie committed:
update PPP

hzxcaryn committed:
Update UG (#228)

* Add GUI section in UG

* Update admin commands phrasing

* Update up to and including employee commands in UG

* Update formatting

* Update figure labels

* Update UG

* Minor tweak to UG

SunBangjie committed:
update manual testing

SunBangjie committed:
add new line at EOF

shanwpf committed:
Update UG screenshots (#232)

* Include screenshots for all shift commands in the UG

* Minor amendments to UG

SunBangjie committed:
Merge pull request #231 from SunBangjie/ppp

minor change in PPP and added Manual Testing

shanwpf committed:
Fix image filenames

SunBangjie committed:
fix undo bug

SunBangjie committed:
Merge pull request #234 from SunBangjie/bugfix

Fix undo bug on storage

shanwpf committed:
Update screenshots and PPP (#233)

* Reduce screenshot height and update PPP

* Minor amendments to UG

waynekoo committed:
V1.5 update images (#235)

* fix minor typo

* update images with border

hzxcaryn committed:
Update PPP & UG (#236)

* Update formatting

* Update DG for timetable

* Update PPP

shanwpf committed:
Update screenshots (#237)

waynekoo committed:
update image with border (#239)

waynekoo committed:
update aboutUs

hzxcaryn committed:
Solve timetable rescaling bug

hzxcaryn committed:
Update UG figure labels (#238)

* Update figure titles

* Update more figure labels in UG

shanwpf committed:
Add table of contents title to DG

hzxcaryn committed:
Add docs

SunBangjie committed:
Merge pull request #240 from hzxcaryn/FixBugs

Solve timetable rescaling bug

SunBangjie committed:
update manual testings

SunBangjie committed:
update use case and user stories

SunBangjie committed:
Merge pull request #242 from SunBangjie/manualTesting

update manual testings, user stories and use cases

hzxcaryn committed:
Update DG and AboutUs (#243)

* Update AboutUs

* Remove irrelevant appendices in DG

* Update jar name and icon

hzxcaryn committed:
Last Changes (#244)

* Update export email

* Update tests

* Collate

* Fix length
